### PR TITLE
CALS-5237 Block Entry of Alpha Characters in Zip Codes for US Addresses

### PR DESCRIPTION
**Pull request by vladimirPe**
Story: https://osi-cwds.atlassian.net/browse/CALS-5237

Description:
The allowCharacters functionality was added to InputComponent (the same as in the InputField)
Zip field in the CommonAddressComponent was changed to allow only numeric characters and the max length set to 5

Tests:
Unit tests were added for InputComponent and CommonAddressComponent

### CODE REVIEW

**Comment by NagaPradeepGajula-osi** [comment]
looking good. Make sure you fix code climate issues. They are pretty small ones.

**Comment by vladimirPe** [comment]
@NagaPradeepGajula-osi  All the code climate bugs are fixed now.
Could someone merge it?
Thanks

**Comment by LeonidMarushevskyi** [comment]
@edgibbs @brunoOSI @ratneshraval @eckdanny-osi @wmitchell-cambria Can you please review and merge PR?

File: src/common/InputField.js [hunk 1]
```diff
@@ -81,4 +81,6 @@ InputField.propTypes = {
   type: PropTypes.string,
   value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
 };
+
+export { sanitizeValue };
```
Comment (wmitchell-cambria) [inline src/common/InputField.js]: Two things:

* It seems really odd to export this shared method from a component file.
* Exporting a previously private method should probably include some unit tests for the method, separate from existing contexts in which it is used. For example, we should probably have a unit test that asserts the current behavior of edge cases like `sanitizeValue(null)`.

**Comment by vladimirPe** [comment]
@wmitchell-cambria, please review the changes

File: src/common/FieldUtils.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const sanitizeValue = (string, allowRegex) => {
```
Comment (edgibbs) [inline src/common/FieldUtils.js]: I realize this function already existed, but given this is being refactored here, a few ideas here:

- Could this just be handled with a String's `#replace` method?
- The null check is supposed to handle developers who explicitly pass no second argument so it's undefined or who pass a null value as the second argument, not sure this is a real scenario and should break other tests they have.
- I think scoping this to removing just alphabetical characters makes more sense, something like `stripNonNumeric()`.

File: src/test/common/FieldUtils_test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { sanitizeValue } from '../../common/FieldUtils';
+
+describe('Field utils', () => {
```
Comment (edgibbs) [inline src/test/common/FieldUtils_test.js]: Tests look fine, but I'd maybe tell more of a story with the describe/it blocks:

```javascript
describe('FieldUtils', () => {
  describe('#santizeValue', () => {
   describe('when value is null', () => {
      it('returns sanatized result', () => {
      });
    });
  });
});

**Comment by edgibbs** [comment]
@wmitchell-cambria earlier comment about the InputField and InputComponent.  There was a lot of work on react-wood-duck in the early days of case management because the business was trying to figure out what they wanted to build.  So we didn't get a lot of input on what was needed for a component library and some of the items may not be as useful as hoped.  @eckdanny-osi is in process of overhauling the whole thing and we should probably make him the default as he's the defacto code owner on this.

**Comment by ratneshraval** [comment]
In CALS we are using third party component to avoid this work ourselves. https://github.com/nosir/cleave.js
We should consider this. Works pretty well. We use it for phone number entry restriction and formatting.

File: src/common/FieldUtils.js (hunk 2 above)
Comment (wmitchell-cambria) [reply src/common/FieldUtils.js]: +1

I'm less concerned about the null check than Ed is, given that `"null".match(null)` and `"anything".match(undefined)"` have potentially counterintuitive behavior. I think I'm leaning the same way as Ed, but I personally don't care as long as whatever behavior you choose is unit tested.

File: src/InputComponent.js [hunk 4]
```diff
@@ -1,16 +1,31 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { sanitizeValue } from './common/FieldUtils';
 
-const InputComponent = props => {
-  let errorMessage = '';
+const extractErrorMessage = props => {
+  let errMessage = '';
```
[the remark this answers is not in the document]
Comment (vladimirPe) [reply src/InputComponent.js]: That was code climate issue, which was fixed. The behavior was not changed.

[the remark this answers is not in the document]
Comment (vladimirPe) [reply src/InputComponent.js]: The Issue was about of complexity level, so I've moved the code out of component scope.
If it is a big concern about a variable name I can change it.

File: src/common/FieldUtils.js (hunk 2 above)
Comment (vladimirPe) [reply src/common/FieldUtils.js]: @edgibbs, 
1. sanitizeValue method was just extracted from InputField component for reusability and was not changed to not regress the code. Refactoring can be done later outside of the scope of this PR.
2. According to @wmitchell-cambria comment earlier I created unit tests to cover corner cases which led to handling situation with null parameters in the method. It is more intuitive to have handled exception instead of unhandled NullPointerException. 
3. The sanitizeValue method is not only for stripping numbers. It can be used for any kind of stripping. So I would prefer to keep the name as is.

Would appreciate if PR will be merged unless there are some critical issues. Now it blocks me from finish my user story in the scope of current sprint. Thanks

Comment (edgibbs) [reply src/common/FieldUtils.js]: I'm not a fan of null checks, and in javascript, you have fun with undefined as well, but I added a story to the CALS backlog to come back and cleanup or replace with cleave.js

https://github.com/nosir/cleave.js

https://osi-cwds.atlassian.net/browse/CALS-6044
